py3status/modules/hueshift.py

```diff
@@ -0,0 +1,204 @@
+    {'full_text': u'blueshift 10000K'}
+]
+
+paused
+{'full_text': u'\[sct, blueshift, redshift\]', 'color': #ff0}
+"""
+
+STRING_NOT_INSTALLED = 'not installed'
+STRING_BAD_COMMAND = 'invalid command'
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_down = 5
+    button_reset = 3
+    button_up = 4
+    command = None
+    delta = 250
+    format = u'[\?color=degraded \[{pgrep}\]|[\?color=hue&show \u263c]' +\
+        '[\?if=is_scrolling {command} {color_temperature}K]]'
+    initial = 6500
+    maximum = 25000
+    minimum = 1000
+    scroll_timeout = 5
+    thresholds = [(6499, '#f3c'), (6500, '#ff3'), (6501, '#3cf')]
+
+    def post_config_hook(self):
+        hueshift_commands = ['sct', 'blueshift', 'redshift']
+        self.pgrep_command = ['pgrep', '-l', '|'.join(hueshift_commands)]
+        if not self.command:
+            self.command = self.py3.check_commands(hueshift_commands)
+        elif self.command not in hueshift_commands:
+            raise Exception(STRING_BAD_COMMAND)
+        elif not self.py3.check_commands(self.command):
+            raise Exception('%s %s' % (self.command, STRING_NOT_INSTALLED))
+        if not self.command:
+            raise Exception(STRING_NOT_INSTALLED)
+
+        self.hue = {
+            'sct': {
+                'set': lambda value: ['sct', str(value)],
+                'reset': lambda value: ['sct']
+            },
+            'blueshift': {
+                'set': lambda value: ['blueshift', '-p', '-t', str(value)],
+                'reset': lambda value: ['blueshift', '-p', '-r'],
+            },
+            'redshift': {
+                'set': lambda value: ['redshift', '-r', '-O', str(value)],
+                'reset': lambda value: ['redshift', '-r', '-x'],
+            },
+        }
+
+        self.index = 0
+        self.is_scrolling = None
+        self.last_command = None
+        self.color_temperature = 6500
+        self._is_running()
+
+        if self.minimum < 1000:
+            self.minimum = 1000
+        if self.maximum > 25000:
+            self.maximum = 25000
+        if self.command == 'sct' and self.maximum > 10000:
+            self.maximum = 10000
+        if not(self.minimum <= self.initial <= self.maximum):
+            self.initial = 6500
+        if not self.pgrep:
+            self._command('set', self.initial)
+
+    def _command(self, action=None, value=None):
+        current_command = self.hue[self.command][action](value)
+        if current_command != self.last_command:
+            self.py3.command_run(current_command)
+            self.last_command = current_command
+        else:
+            self.py3.prevent_refresh()
+
+    def _is_running(self):
+        try:
+            ps = self.py3.command_output(self.pgrep_command).splitlines()
+            names = []
+            for line in ps:
+                names.append(line.split()[-1])
+            self.pgrep = ', '.join(names)
```

Comment:
Now you coding is improving I'm going to add new things.

This is really bad. you method is `_is_running()` so I'm expecting it to return a boolean which it does.  However it is also setting some stuff here.  So when I read the code this will screw me up big time because it is soing this stuff when it sounds like it doesn't.

You could change it to `_get_running_commands()` or something and just return the list (empty if none) and do the to string list if needed later.

Reply:
I had couple of `is_running` earlier... but then I saw that I could get rid of them (few lines) and use `self.pgrep` as it's 100% doing the same thing, so I removed them.